static/src/js/components/Facets/__tests__/Facets.test.js

```diff
@@ -247,245 +313,283 @@ describe('Facets component', () => {
-    expect(facetsGroup.props().facet.hasChildren).toEqual(true)
-    expect(facetsGroup.props().facet.title).toEqual('Tiling System')
-    expect(facetsGroup.props().facet.totalSelected).toEqual(0)
-    expect(facetsGroup.props().facet.children).toEqual([{
-      applied: false,
-      count: 1,
-      has_children: false,
-      links: { apply: 'http://example.com/apply_tiling_system_link' },
-      title: 'Mock Tiling System Facet',
-      type: 'filter'
-    }])
+  test('renders tiling system FacetsGroup', async () => {
+    const { renderContainer, props } = setup()
+
+    const { container } = renderContainer(props)
+
+    const testIndex = 7
+
+    const facetGroupText = Object.keys(props.facetsById)[testIndex]
+
+    const facetGroup = screen.getByTestId(`facet_group-${kebabCase(facetGroupText)}`)
+
+    expect(facetGroup).toBeInTheDocument()
+    expect(screen.queryByTestId(`facet-${kebabCase(facetGroupText)}`)).toBeNull()
+
+    const user = userEvent.setup()
+    // clicking on the dropdown to show the different facet items
+    await user.click(getAllByRole(container, 'button').at(testIndex + 1))
+    expect(screen.getByTestId(`facet-${kebabCase(facetGroupText)}`)).toBeInTheDocument()
+    // clicking again and making sure the facet items go away
+    await user.click(getAllByRole(container, 'button').at(testIndex + 1))
+    expect(screen.queryByTestId(`facet-${kebabCase(facetGroupText)}`)).toBeNull()
   })
 
-  test('renders horizontal data resolution FacetsGroup', () => {
-    const { enzymeWrapper } = setup()
-
-    const facetsGroup = enzymeWrapper.find(FacetsGroup).at(9)
-
-    expect(facetsGroup.props().facet.applied).toEqual(false)
-    expect(facetsGroup.props().facet.autocompleteType).toEqual('horizontal_data_resolution')
-    expect(facetsGroup.props().facet.hasChildren).toEqual(true)
-    expect(facetsGroup.props().facet.title).toEqual('Horizontal Data Resolution')
-    expect(facetsGroup.props().facet.totalSelected).toEqual(0)
-    expect(facetsGroup.props().facet.children).toEqual([{
-      applied: false,
-      count: 1,
-      has_children: false,
-      links: { apply: 'http://example.com/apply_horizontal_data_resolution_link' },
-      title: 'Mock Horizontal Data Resolution Facet',
-      type: 'filter'
-    }])
+  test('renders horizontal data resolution FacetsGroup', async () => {
+    const { renderContainer, props } = setup()
+
+    const { container } = renderContainer(props)
+
+    const testIndex = 8
+
+    const facetGroupText = Object.keys(props.facetsById)[testIndex]
+
+    const facetGroup = screen.getByTestId(`facet_group-${kebabCase(facetGroupText)}`)
+
+    expect(facetGroup).toBeInTheDocument()
+    expect(screen.queryByTestId(`facet-${kebabCase(facetGroupText)}`)).toBeNull()
+
+    const user = userEvent.setup()
+    // clicking on the dropdown to show the different facet items
+    await user.click(getAllByRole(container, 'button').at(testIndex + 1))
```

Comment:
```suggestion
    const facetButton getAllByRole(container, 'button').at(testIndex + 1)
```
set to var with nice name and reuse